public/legacy/include/SugarFolders/SugarFolders.php

```diff
@@ -876,25 +884,19 @@ public function getFoldersForSettings($focusUser = null)
                 if (isset($a['dynamic_query'])) {
                     unset($a['dynamic_query']);
                 }
-
-                if ($a['is_group'] == 1) {
+                //add check to remove deleted folders
+                if ($a['is_group'] == 1 && $a['deleted'] != 1) {
```

Comment:
in between 888 and 889 need to add: 
`if (!in_array($a['id'], $groupFolderIds)) {`

How do I do that now it's in PR?